src/components/modals/ManagePlanDerivationGroupsModal.svelte

```diff
@@ -0,0 +1,315 @@
+        filter: 'string',
+        headerName: 'Source type',
+        resizable: true,
+        sortable: true,
+        suppressAutoSize: false,
+        suppressSizeToFit: false,
+      },
+      {
+        field: 'derived_event_total',
+        filter: 'number',
+        headerName: 'Derived Events in Derivation Group',
+        sortable: true,
+        suppressAutoSize: true,
+        suppressSizeToFit: true,
+        valueFormatter: params => {
+          return params?.value.length;
+        },
+        width: 250,
+      },
+      {
+        cellDataType: 'boolean',
+        editable: true,
+        headerName: 'Included in Plan',
+        resizable: false,
+        suppressAutoSize: true,
+        suppressSizeToFit: true,
+        valueGetter: (params: ValueGetterParams<DerivationGroup>) => {
+          const { data } = params;
+          if (data) {
+            return $selectedPlanDerivationGroupNames.includes(data.name);
+          }
+          return false;
+        },
+        width: 115,
+      },
+      {
+        cellClass: 'action-cell-container',
+        cellRenderer: (params: DerivationGroupCellRendererParams) => {
+          const actionsDiv = document.createElement('div');
+          actionsDiv.className = 'actions-cell';
+          new DataGridActions({
+            props: {
+              rowData: params.data,
+              viewCallback: params.viewDerivationGroup,
+              viewTooltip: {
+                content: 'View Derivation Group',
+                placement: 'bottom',
+              },
+            },
+            target: actionsDiv,
+          });
+
+          return actionsDiv;
+        },
+        cellRendererParams: {
+          viewDerivationGroup,
+        } as CellRendererParams,
+        headerName: '',
+        resizable: false,
+        sortable: false,
+        suppressAutoSize: true,
+        suppressSizeToFit: true,
+        width: 40,
+      },
+    ];
+  }
+
+  function viewDerivationGroup(viewedDerivationGroup: DerivationGroup) {
+    const derivationGroup = $derivationGroups.find(
+      derivationGroup => derivationGroup.name === viewedDerivationGroup.name,
+    );
+    selectedDerivationGroup = derivationGroup;
+  }
+
+  function getRowId(derivationGroup: DerivationGroup): string {
+    return `${derivationGroup.name}:${derivationGroup.source_type_name}`;
+  }
+
+  async function changeDerivationGroupAssociation(checked: boolean, derivationGroupName: string | undefined) {
+    if (derivationGroupName !== undefined) {
+      if (checked) {
+        await effects.insertDerivationGroupForPlan(derivationGroupName, $plan, user);
+      } else {
+        await effects.deleteDerivationGroupForPlan(derivationGroupName, $plan, user);
+      }
+    }
+  }
+</script>
+
+<Modal height={600} width={1000}>
+  <ModalHeader on:close>Manage Derivation Groups</ModalHeader>
+  <ModalContent style=" overflow-y:scroll; padding:0;">
+    <CssGrid columns={modalColumnSize} minHeight="100%">
```

Comment:
Instead of using `overflow-y:scroll` in various places (not generally recommended since it forces a scrollbar to appear when not needed) can you pass a class to CssGrid that gives it 100% height and then set overflow to auto on the body of the modal content (or make a wrapper around the content within and give it the overflow auto)?

Reply:
Removed & matched the given styling to `UploadViewModal.svelte`!